org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/internal/gradle/checksums/WrapperValidator.java

```diff
@@ -0,0 +1,97 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Red Hat Inc. and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Red Hat Inc. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jdt.ls.internal.gradle.checksums;
+
+import java.io.BufferedReader;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.net.URLConnection;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.NoSuchAlgorithmException;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.ls.core.internal.JavaLanguageServerPlugin;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+
+public class WrapperValidator {
+
+	private static final String DISTRIBUTION_URL = "distributionUrl";
+	private static final String GRADLE_WRAPPER_PROPERTIES = "gradle/wrapper/gradle-wrapper.properties";
+	private static final String GRADLE_WRAPPER_JAR = "gradle/wrapper/gradle-wrapper.jar";
+	private final boolean DISABLED = Boolean.getBoolean("gradle.wrapper.check.disabled");
+	private HashProvider hashProvider;
+	protected static Cache<String, String> downloadRequestsCache = CacheBuilder.newBuilder().maximumSize(100).expireAfterWrite(1, TimeUnit.HOURS).build();
+
+
+	public WrapperValidator() {
+		this(new HashProvider());
+	}
+
+	public WrapperValidator(HashProvider hashProvider) {
+		this.hashProvider = hashProvider;
+	}
+
+	public String checkWrapper(String baseDir) throws IOException, NoSuchAlgorithmException {
+		if (DISABLED) {
+			return null;
+		}
+		Path wrapperJar = Paths.get(baseDir, GRADLE_WRAPPER_JAR);
+		if (!wrapperJar.toFile().exists()) {
+			return wrapperJar.toString() + " doesn't exist.";
+		}
+		Path wrapperProperties = Paths.get(baseDir, GRADLE_WRAPPER_PROPERTIES);
+		if (!wrapperProperties.toFile().exists()) {
+			return wrapperProperties.toString() + " doesn't exist.";
+		}
+		try (InputStream is = new FileInputStream(wrapperProperties.toFile())) {
+			Properties props = new Properties();
+			props.load(is);
+			String distributionUrl = props.getProperty(DISTRIBUTION_URL);
+			if (distributionUrl == null) {
+				return "Invalid the distributionUrl property";
+			}
+			String sha256Url = distributionUrl.replace("-bin.zip","").replace("-all.zip", "") + "-wrapper.jar.sha256";
```

Comment:
no. we should instead check the checksum exists in :
1- a cache version of https://services.gradle.org/versions/all
2- a live version of https://services.gradle.org/versions/all, if it wasn't found in the cache

This is fairly similar to https://github.com/gradle/wrapper-validation-action/blob/master/src/checksums.ts, but with a local cache

Reply:
Fixed.